gguf-py/tests/test_gguf.py

```diff
@@ -1,7 +1,48 @@
-import gguf
+import os
+import tempfile
+import pytest
+import numpy as np
+from gguf import GGUFWriter, GGMLQuantizationType, GGUFValueType, MODEL_ARCH
 
-# TODO: add tests
+# Fixture to create a temporary directory for testing
+@pytest.fixture
+def temp_dir():
+    with tempfile.TemporaryDirectory() as tempdir:
+        yield tempdir
 
+# Test GGUFWriter class
+def test_gguf_writer(temp_dir):
+    # Create a GGUFWriter instance
+    writer = GGUFWriter(os.path.join(temp_dir, "test.gguf"), "gpt2")
 
-def test_write_gguf():
-    pass
+    # Add metadata
+    writer.add_name("Test Model")
+    writer.add_author("John Doe")
+    writer.add_description("A test model")
+    writer.add_architecture()
+    writer.add_tensor_data_layout("NHWC")
+    writer.add_context_length(512)
+
+    # Add a tensor
+    tensor_data = np.random.rand(10, 10).astype(np.float32)
+    writer.add_tensor("test_tensor", tensor_data)
+
+    # Write to file and close
+    writer.write_header_to_file()
+    writer.write_tensors_to_file()
+    writer.close()
+
+    # Verify the generated file exists
+    assert os.path.exists(os.path.join(temp_dir, "test.gguf"))
+
+# Test GGMLQuantizationType
+def test_ggml_quantization_type():
+    assert GGMLQuantizationType.F32 == 0
+    assert GGMLQuantizationType.F16 == 1
+    assert GGMLQuantizationType.Q4_0 == 2
+
+# Test GGUFValueType
+def test_gguf_value_type():
```

Comment:
This test is also redundant. There's no need to write test cases for enums / constants.
A better test case would be to read the written GGUF file, but there's no reader class implemented in Python yet. So we can verify the checksum of the file instead.